api/src/schemas.py

```diff
@@ -203,10 +203,76 @@ def is_accessible(cls, values):
 
 
 class CmrInput(WorkflowInputBase):
-    # cmr discovery
-    discovery: Literal[Discovery.cmr]
-
+    discovery: Literal["cmr"]
     version: Optional[str]
     include: Optional[str]
     temporal: Optional[List[datetime]]
     bounding_box: Optional[List[float]]
+
+
+# allows the construction of models with a list of discriminated unions
+ItemUnion = Annotated[Union[S3Input, CmrInput], Field(discriminator="discovery")]
+
+
+class Dataset(BaseModel):
+    collection: str
+    title: str
+    description: str
+    license: str
+    dashboard_is_periodic: bool
+    dashboard_time_density: str
+    spatial_extent: BboxExtent
+    temporal_extent: TemporalExtent
+    sample_files: List[str]  # unknown how this will work with CMR
+    discovery_items: List[ItemUnion]
+
+    class Config:
+        extra = Extra.allow
+
+    @root_validator
+    def check_time_density(cls, v):
+        if v["dashboard_is_periodic"] and v["dashboard_time_density"] not in [
+            "month",
+            "day",
+            "year",
+        ]:
+            raise ValueError("Invalid time density")
```

Comment:
maybe we can add some more information to make it easier for user to fix the issue. something that says: when is_periodic is set to true, time_density is required and it should be one of these values : ....